.github/workflows/codeql.yml

```diff
@@ -163,44 +63,114 @@ jobs:
 
             # C++ Build
             - name: Build C++ components
-              if: matrix.language == 'c-cpp' && env.SHOULD_RUN == 'true'
+              if: matrix.language == 'c-cpp'
               shell: bash
               run: |
                   cd glide-core
                   cargo build --release
 
+            # Go Build
+            - name: Set up Go cache
+              if: matrix.language == 'go'
+              uses: actions/cache@v4
+              with:
+                  path: |
+                      ~/.cache/go-build
+                      ~/go/pkg/mod
+                      ~/go/bin
+                  key: ${{ runner.os }}-go-${{ hashFiles('go/go.mod', 'go/go.sum') }}-protoc-gen-go-v1.33.0
+                  restore-keys: |
+                      ${{ runner.os }}-go-
+
+            - name: Cache protoc binary
+              if: matrix.language == 'go'
+              uses: actions/cache@v4
+              with:
+                  path: /usr/local/bin/protoc
+                  key: ${{ runner.os }}-protoc-29.1
+                  restore-keys: |
+                      ${{ runner.os }}-protoc-
+
+            - name: Install protoc-gen-go
+              if: matrix.language == 'go'
+              shell: bash
+              run: |
+                  # Check if protoc-gen-go is already installed
+                  if ! command -v protoc-gen-go &> /dev/null; then
+                      echo "Installing protoc-gen-go..."
+                      go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.33.0
+                  else
+                      echo "protoc-gen-go already installed: $(protoc-gen-go --version)"
+                  fi
+
+            - name: Install protoc compiler
+              if: matrix.language == 'go'
+              shell: bash
+              run: |
+                  # Check if protoc is already installed with correct version
+                  if ! command -v protoc &> /dev/null || ! protoc --version | grep -q "29.1"; then
+                      echo "Installing protoc 29.1..."
+                      PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+                      curl -LO $PB_REL/download/v29.1/protoc-29.1-linux-x86_64.zip
+                      sudo unzip protoc-29.1-linux-x86_64.zip -d /usr/local
+                      sudo chmod +x /usr/local/bin/protoc
+                  else
+                      echo "protoc already installed: $(protoc --version)"
+                  fi
```

Comment:
I think this two could be merged into one step, with a condition like `go && !cache.hit`

Reply:
BTW re-install protoc could be easier and faster than cache restore. It is just a single file.
```yml
        - name: Install protoc (protobuf)
          uses: arduino/setup-protoc@v3
          with:
              version: "29.1"
              repo-token: ${{ inputs.github-token }}
```